memtier_benchmark.cpp

```diff
@@ -839,6 +847,7 @@ void usage() {
             "      --key=FILE                 Use specified private key for TLS\n"
             "      --cacert=FILE              Use specified CA certs bundle for TLS\n"
             "      --tls-skip-verify          Skip verification of server certificate\n"
+            "      --sni=STRING               Add an SNI header\n"
```

Comment:
Any reason to toggle this on/off? Why not always add it when doing TLS?

Reply:
@ushachar I think that'll work for the mainstream cases, but there's probably a long tail of cases where you'd want a custom name, specify an IP as a hostname + SNI (e.g. no DNS resolution), etc.

We *could* consider to use the hostname by default (if it's a name and not an IP) if we still have the option to control that manually.